internal/git/provider/github_provider.go

```diff
@@ -53,6 +55,29 @@ func (p GitHubProvider) AddCommentToPullRequest(ctx context.Context, pr PullRequ
 	}, nil
 }
 
+func (p *GitHubProvider) GetLastComment(ctx context.Context, pr PullRequest) (*Comment, error) {
+	// TODO make sure that we get the last comment
+	comments, _, err := p.client.Issues.ListComments(ctx, pr.Repository.String(), pr.Number, &scm.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list pull requests: %w", err)
+	}
+
+	if len(comments) == 0 {
+		return nil, nil
+	}
+
+	sort.Slice(comments, func(i, j int) bool {
+		return comments[i].Created.After(comments[j].Created)
+	})
+
+	latestComment := comments[0]
```

Comment:
I think it should check timestamp or something. Checking the last comment has a lot of flaws.
1. Read the same `!replan` again (mitigated with posting an "in progress" comment here)
2. If someone writes `!replan` and then someone makes a comment before the next polling interval occurs, it will not be re-planned with no reason provided (not even in logs). In dev environment it's not a big deal, but in production they may set the polling to 2-5 minutes (api limitations, stress on the cluster), which is pretty long to check only the last comment.

This can be a `ListComments` that returns with all comments sorted by creation time and the consumer can decide what to do.

On consumer side they can use for example the last plan output time as an anchor point.

Reply:
"sorted by creation" = descending order, we need the last `!replan`.